tests/datasets/test_l8biome.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import pytest
+import torch
+import torch.nn as nn
+from _pytest.fixtures import SubRequest
+from _pytest.monkeypatch import MonkeyPatch
+from rasterio.crs import CRS
+
+from torchgeo.datasets import BoundingBox, IntersectionDataset, L8Biome, UnionDataset
+
+class TestL8Biome:
+    @pytest.fixture(
+        params=[
+            ["SR_B1", "SR_B2", "SR_B3", "SR_B4", "SR_B5", "SR_B6", "SR_B7"],
+            ["SR_B4", "SR_B3", "SR_B2", "SR_QA_AEROSOL"],
+        ]
+    )
+    def dataset(self, monkeypatch: MonkeyPatch, request: SubRequest) -> L8Biome:
+        root = os.path.join("tests", "data", "l8biome")
+        bands = request.param
+        transforms = nn.Identity()
+        return L8Biome(root, bands=bands, transforms=transforms)
+
+    def test_separate_files(self, dataset: L8Biome) -> None:
+        assert dataset.index.count(dataset.index.bounds) == 1
+
```

Comment:
This is somewhat specific to the Landsat dataset, you don't have to copy it for your L8 Biome dataset.